File: backend/app/npc/random_ai.py
import random
from ..models import Hand, IdoHand, Direction


class RandomAI:
    """ランダムに手を選ぶAI"""

    def __init__(self, npc_name: str):
        """初期化

        Args:
            npc_name: NPCの名前
        """
        self.name = npc_name

    def choose_hand(self, rule_type: str) -> Hand | IdoHand:
        """じゃんけんの手を選択

        Args:
            rule_type: ゲームルールタイプ

        Returns:
            Hand | IdoHand: 選択した手
        """
        if rule_type == "ido_janken":
            return random.choice(["rock", "paper", "scissors", "well"])
        else:
            return random.choice(["rock", "paper", "scissors"])

    def choose_direction(self) -> Direction:
        """あっちむいてホイの方向を選択

        Returns:
            Direction: 選択した方向
        """
        return random.choice(["up", "down", "left", "right"])

File: src/game/npc/RandomAI.ts
// Easy 難易度: 完全ランダム NPC (Issue #6)。
// 3 手 / 4 手 / 4 方向すべて等確率で選ぶ。最弱・初心者向け。

import type { Direction, GameRuleType, Hand, IdoHand } from '../types'
import { AIRandomSource, NpcAI, OpponentMove, defaultRandom } from './types'

const HANDS_3: Hand[] = ['rock', 'paper', 'scissors']
const HANDS_4: IdoHand[] = ['rock', 'paper', 'scissors', 'well']
const DIRS: Direction[] = ['up', 'down', 'left', 'right']

export class RandomAI implements NpcAI {
  readonly difficulty = 'easy' as const
  private readonly random: AIRandomSource

  constructor(random: AIRandomSource = defaultRandom) {
    this.random = random
  }

  chooseHand(
    rule: GameRuleType,
    _opponentHistory?: OpponentMove[]
  ): Hand | IdoHand {
    void _opponentHistory
    if (rule === 'ido_janken') {
      const idx = Math.floor(this.random.next() * HANDS_4.length)
      return HANDS_4[Math.min(idx, HANDS_4.length - 1)]
    }
    const idx = Math.floor(this.random.next() * HANDS_3.length)
    return HANDS_3[Math.min(idx, HANDS_3.length - 1)]
  }

  chooseDirection(_opponentHistory?: OpponentMove[]): Direction {
    void _opponentHistory
    const idx = Math.floor(this.random.next() * DIRS.length)
    return DIRS[Math.min(idx, DIRS.length - 1)]
  }
}
